lib/datastore.js

```diff
@@ -304,17 +304,11 @@ class DataStore {
       throw new TypeError("expected item");
     }
 
-    // TODO: check for existing item?
-    // TODO: deep copy
-    let id = UUID(),
-        now = new Date();
+    // validate, and fill defaults into, {item}
+    item = Items.prepare(item);
 
-    item = Object.assign({}, item);
-    item.id = id;
-    item.created = item.created || now;
-    item.modified = item.modified || now;
-
-    let active = !item.disabled ? "active" : "",
+    let id = item.id,
+        active = !item.disabled ? "active" : "",
         encrypted = await self.keystore.encrypt(item);
```

Comment:
It looks like the encrypted data now contains the item's UUID when it didn't before. Is that what you want?

Reply:
the `item` has always contained the `id`.